config/roles.php

```diff
@@ -4,237 +4,171 @@
 use App\Site\Models\Role;
 
 $rootAssignable = [
-    Role::FOUNDER,
-    Role::ARCHITECT,
-    Role::RELEASE_MANAGER,
     Role::ADMINISTRATOR,
+    Role::ARCHITECT,
     Role::BETA,
+    Role::ENGINEER,
+    Role::FOUNDER,
+    Role::RELEASE_MANAGER,
 ];
 
 $adminAssignable = [
```

Comment:
As an admin, assigning a valid role to another admin (like Developer) causes the other admin to lose their admin role.

Reply:
fixed in https://github.com/RetroAchievements/RAWeb/pull/2165/commits/7790b8f44921b44e78253b72ab0acd78e2079af8 - this was not quite finished yet

Reply:
As a root user, I added admin to another user and was no longer able to add or remove roles from that user. As such, I'm unable to test the original problem.

I see the same behavior as a non-root admin (cannot modify another admin), but I'd expect root to be able to.

I would also expect a non-root admin be able to add/remove lesser roles on another admin.

Reply:
I'm still seeing this behavior.

A Root user can add the Administrator role to another user, but once that has occurred, the root user cannot modify that user at all. The Administrator role cannot be removed, and no other roles can be added or removed.

Similarly, a non-Root Administrator cannot add or remove any roles from another non-Root Administrator.

Reply:
fix root role privileges fro admins in https://github.com/RetroAchievements/RAWeb/pull/2165/commits/a5a889fc17518e487cee500ea213fa6e85fdcd5c

non-Root Administrator to non-Root Administrator should not be necessary as admins can edit their own roles at any time. however, slightly improved in https://github.com/RetroAchievements/RAWeb/pull/2165/commits/e1760a66f351ea1a543038a48142d4e9a6d5a495